pkg/kademlia/kademlia.go

```diff
@@ -280,37 +283,33 @@ func (k *Kademlia) FindNode(ctx context.Context, nodeID storj.NodeID) (_ pb.Node
 	}
 	defer k.lookups.Done()
 
-	return k.lookup(ctx, nodeID, false)
+	results, err := k.lookup(ctx, nodeID)
+	if err != nil {
+		return pb.Node{}, err
+	}
+	if len(results) < 1 {
+		return pb.Node{}, NodeNotFound.New("")
+	}
+	return *results[0], nil
 }
 
 //lookup initiates a kadmelia node lookup
-func (k *Kademlia) lookup(ctx context.Context, nodeID storj.NodeID, isBootstrap bool) (_ pb.Node, err error) {
+func (k *Kademlia) lookup(ctx context.Context, nodeID storj.NodeID) (_ []*pb.Node, err error) {
 	defer mon.Task()(&ctx)(&err)
 	if !k.lookups.Start() {
-		return pb.Node{}, context.Canceled
+		return nil, context.Canceled
 	}
 	defer k.lookups.Done()
 
-	kb := k.routingTable.K()
-	var nodes []*pb.Node
-	if isBootstrap {
```

Comment:
We don't need to tell it to distinguish between bootstrap or not?

Reply:
the only time bootstrap nodes matter is when your routing table is empty, and only then to bootstrap the lookup. as soon as you have any nodes in your routing table or lookup queue, the normal kademlia operation is totally sufficient. we don't really need other logic.

the key thing in this pr for getting the bootstrap nodes into the mix is making sure we call the transport observers with their IDs, which gets them added to routing tables, node databases, etc.